src/genopets/infos.ts

```diff
@@ -0,0 +1,233 @@
+import {
+  Connection,
+  PublicKey,
+  AccountInfo,
+  DataSizeFilter,
+  MemcmpFilter,
+  GetProgramAccountsConfig,
+} from "@solana/web3.js";
+import { IFarmInfoWrapper, IInstanceFarm } from "../types";
+import { GENOPETS_FARM_PROGRAM_ID } from "./ids";
+import { DEPOSIT_LAYOUT, FARMER_LAYOUT, FARM_LAYOUT } from "./layouts";
+import * as types from ".";
+import { BN } from "bn.js";
+import { getMultipleAccounts } from "../utils";
+
+let infos: IInstanceFarm;
+infos = class InstanceGenopets {
+  static async getAllFarms(connection: Connection): Promise<types.FarmInfo[]> {
+    const farmId = PublicKey.findProgramAddressSync([Buffer.from("stake-master-seed")], GENOPETS_FARM_PROGRAM_ID)[0];
+    const farmAccountInfo = await connection.getAccountInfo(farmId);
+    const farm = this.parseFarm(farmAccountInfo?.data!, farmId);
+
+    return [farm];
+  }
+
+  static async getAllFarmWrappers(connection: Connection): Promise<types.FarmInfoWrapper[]> {
+    return (await this.getAllFarms(connection)).map((farmInfo) => new FarmInfoWrapper(farmInfo));
+  }
+
+  static async getFarm(connection: Connection, farmId: PublicKey): Promise<types.FarmInfo> {
+    const expectedFarmId = PublicKey.findProgramAddressSync(
```

Comment:
1. Remove `findProgramAddress` and `farmId` check
2. Catch the case when farm is null